### ISSUE TITLE
`this` in static methods has unknown type

### ISSUE THREAD

**Issue by rictic** [google/closure-compiler#3340]
Compiling this code will fail the BanUnknownThis JS conformance check

```javascript
class Foo {
  static method() {
    console.log(this);
  }
}
```

Related issue: #2397

(upstreaming b/130821356 that I filed internally)

**Comment by rictic** [comment]
As a workaround, casting the `this` to `{typeof Foo}` appears to work.

**Comment by rictic** [comment]
It looks like this was fixed by 5b1e5316e5228d6d9f205e28440d67d9ec007f4b
